pkg/controller/elasticsearch/filesettings/secret_test.go

```diff
@@ -0,0 +1,208 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package filesettings
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	commonv1 "github.com/elastic/cloud-on-k8s/v2/pkg/apis/common/v1"
+	policyv1alpha1 "github.com/elastic/cloud-on-k8s/v2/pkg/apis/stackconfigpolicy/v1alpha1"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/utils/k8s"
+)
+
+func Test_newSettingsSecret(t *testing.T) {
+	es := types.NamespacedName{
+		Namespace: "esNs",
+		Name:      "esName",
+	}
+	policy := policyv1alpha1.StackConfigPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "policyNs",
+			Name:      "policyName",
+		},
+		Spec: policyv1alpha1.StackConfigPolicySpec{
+			Elasticsearch: policyv1alpha1.ElasticsearchConfigPolicySpec{
+				ClusterSettings: &commonv1.Config{Data: map[string]interface{}{"a": "b"}},
+			},
+		},
+	}
+
+	// no policy
+	ss, err := NewSettingsSecret(nil, es, nil)
+	assert.NoError(t, err)
+	assert.Equal(t, "esNs", ss.Namespace)
+	assert.Equal(t, "esName-es-file-settings", ss.Name)
+
+	// policy
+	ss, err = NewSettingsSecret(nil, es, &policy)
+	assert.NoError(t, err)
+	assert.Equal(t, "esNs", ss.Namespace)
+	assert.Equal(t, "esName-es-file-settings", ss.Name)
```

Comment:
Test that settings bytes are not `nil`?

Reply:
Sure, it's like I haven't finish the work here.